### PR TITLE
Clarify cuprite driver configuration

### PR DESCRIPTION
**Pull request by mkllnk**
#### What? Why?

Rails is registering a driver called `cuprite`. And when we did that as well the driver got registered three times somehow. It looked like our driver options were used in the end but just to clarify I gave it a unique name.

This was inspired by:

* https://github.com/ViewComponent/view_component/pull/1877

It suggests that it may avoid dead browser errors on CI. We'll see. It could close:

- #11669
- #11027
- #10220

#### What should we test?

- Specs only. Maybe run them a few times?

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Review by dacook — APPROVED** [review]
It looks like a workaround; I wonder if there's a better way. Why would Rails be overwriting it, or registering "cuprite" in the first place? Maybe it happens automatically when the gem is present. 

🤷  If it makes the specs work better than I'm all for it! 👍

**Review by rioug — APPROVED** [review]
Nice :+1:  :crossed_fingers: it helps

**Comment by rioug** [comment]
I re ran the specs, @dacook also did it, so if it passes I guess it's ready to be merged

**Comment by mkllnk** [comment]
@dacook 
> I wonder if there's a better way. Why would Rails be overwriting it, or registering "cuprite" in the first place?

I was wondering that as well. And I found the code doing it in Actionpack:

https://github.com/rails/rails/blob/253cddd92d7dc7ed170691c173d079073efdc3c2/actionpack/lib/action_dispatch/system_testing/driver.rb#L36-L47

Interestingly, that is even initialised when starting a development console. I thought that we could just pass the options on to the `driven_by` directive but it didn't quite work and the results were more confusing than before.

Basically, when we use a driver called `:cuprite` then Rails is doing its magic and doesn't expect anyone else to interfere with it here. Using our own name is therefore the better option when not using Rails magic.
